torch/csrc/jit/codegen/cuda/type.cpp

```diff
@@ -8,6 +8,62 @@ namespace jit {
 namespace fuser {
 namespace cuda {
 
+bool isFloatingPointType(DataType dtype) {
+  switch (dtype) {
+    case DataType::Bool:
+      return false;
+    case DataType::Double:
+      return true;
+    case DataType::Float:
+      return true;
+    case DataType::Half:
+      return true;
+    case DataType::Int:
+      return false;
+    case DataType::Null:
+      TORCH_CHECK(
+          false, "Null type is not a valid argument to isFloatingPoint");
+    default:
+      TORCH_CHECK(false, "Type not supported in isFloatingPoint");
+  }
+}
+
+bool isIntegralType(DataType dtype) {
+  switch (dtype) {
+    case DataType::Bool:
+      return false;
+    case DataType::Double:
+      return false;
+    case DataType::Float:
+      return false;
+    case DataType::Half:
+      return false;
+    case DataType::Int:
+      return true;
+    case DataType::Null:
+      TORCH_CHECK(
+          false, "Null type is not a valid argument to isFloatingPoint");
+    default:
+      TORCH_CHECK(false, "Type not supported in isFloatingPoint");
+  }
+}
+
+bool isIntegerOp(const BinaryOpType bopt) {
+  return bopt >= BinaryOpType::Mod && bopt <= BinaryOpType::Xor;
+}
+
+bool isLogicalOp(const BinaryOpType bopt) {
+  return bopt >= BinaryOpType::Eq && bopt <= BinaryOpType::NE;
+}
+
+bool maybeBooleanOperator(const BinaryOpType bopt) {
```

Comment:
why 'maybe' instead of 'is'?

Reply:
I guess "also" would be a reasonable interpretation, will change to `alsoBooleanOperator` it's strange that PyTorch doesn't have a separate `and` from `bitwise_and` that they actually use.